lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java

```diff
@@ -44,19 +63,61 @@ public YangInstanceIdentifier serialize(final String identifier) {
             data = data.substring(0, data.length() - 1);
         }
         final List<String> pathArgs = Arrays.asList(data.split("/"));
-        final String[] first = pathArgs.get(0).split(":");
-        final String moduleName = first[0];
-        final Optional<? extends Module> module = this.schemaContext.findModule(moduleName);
-        if (! module.isPresent()) {
-            throw new IllegalStateException("Module with name" + moduleName + " not found");
+        final String[] nameRevision_pathStart = pathArgs.get(0).split(":");
+        final String[] name_Revision = nameRevision_pathStart[0].split("@");
+        final String moduleName = name_Revision[0];
+        String revisionString = null;
+        if (name_Revision.length > 1) {
+            revisionString = name_Revision[1];
         }
-        final QNameModule qNameModule = module.get().getQNameModule();
-        pathArgs.set(0, first[1]);
+        Optional<Revision> requestedRevision = Optional.empty();
+        try {
+            requestedRevision = Revision.ofNullable(revisionString);
+        } catch (DateTimeException e) {
+            LOG.warn("Wrongly formatted revision detected {} ", revisionString);
```

Comment:
Rather return error and stop executing method. If wrongly formatted, user can "quietly" get another YIID than was expected and it may cause unpleasant and hard to find errors later.